iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/extractor/mqtt/MQTTPublishHandler.java

```diff
@@ -0,0 +1,374 @@
+          return;
+        }
+        pendingQueue.waitedOffer(event);
+      }
+    } catch (Exception e) {
+      LOGGER.warn(
+          "meet error when polling mqtt source device {}, measurements {}, at time {}, because ",
+          message.getDevice(),
+          message.getMeasurements(),
+          message.getTimestamp(),
+          e);
+    }
+  }
+
+  private EnrichedEvent generateEvent(final TreeMessage message) throws QueryProcessException {
+    final String deviceId = message.getDevice();
+    final List<String> measurements = message.getMeasurements();
+    final long[] timestamps = new long[] {message.getTimestamp()};
+    final MeasurementSchema[] schemas = new MeasurementSchema[measurements.size()];
+    final TSDataType[] dataTypes =
+        message.getDataTypes() == null
+            ? new TSDataType[message.getMeasurements().size()]
+            : message.getDataTypes().toArray(new TSDataType[0]);
+    final List<String> values = message.getValues();
+    final BitMap[] bitMaps = new BitMap[schemas.length];
+    if (message.getDataTypes() == null) {
+      for (int index = 0; index < values.size(); ++index) {
+        dataTypes[index] = TypeInferenceUtils.getPredictedDataType(values.get(index), true);
+      }
+    }
+
+    for (int i = 0; i < schemas.length; i++) {
+      schemas[i] = new MeasurementSchema(measurements.get(i), dataTypes[i]);
+    }
+    final Object[] inferredValues = new Object[values.size()];
+
+    // For each measurement value, parse it and then wrap it in a one-dimensional array.
+    for (int i = 0; i < values.size(); ++i) {
+      Object parsedValue = CommonUtils.parseValue(dataTypes[i], values.get(i));
+      // Wrap the parsed value into a one-dimensional array based on its type.
+      if (parsedValue instanceof Integer) {
+        inferredValues[i] = new int[] {(Integer) parsedValue};
+      } else if (parsedValue instanceof Long) {
+        inferredValues[i] = new long[] {(Long) parsedValue};
+      } else if (parsedValue instanceof Float) {
+        inferredValues[i] = new float[] {(Float) parsedValue};
+      } else if (parsedValue instanceof Double) {
+        inferredValues[i] = new double[] {(Double) parsedValue};
+      } else if (parsedValue instanceof Boolean) {
+        inferredValues[i] = new boolean[] {(Boolean) parsedValue};
+      } else if (parsedValue instanceof String) {
+        inferredValues[i] = new String[] {(String) parsedValue};
+      } else if (parsedValue instanceof Binary) {
+        inferredValues[i] = new Binary[] {(Binary) parsedValue};
+      } else {
+        // For any other type, wrap it as an Object array.
+        inferredValues[i] = new Object[] {parsedValue};
+      }
+    }
+
+    final Tablet eventTablet =
+        new Tablet(deviceId, Arrays.asList(schemas), timestamps, inferredValues, bitMaps, 1);
+
+    return new PipeRawTabletInsertionEvent(
+        false,
+        deviceId,
+        null,
+        null,
+        eventTablet,
+        true,
```

Comment:
This may better be set to "false" in tree model to be consistent with the normal MQTT writing. It is in table model that it shall be set to true....